src/currentprocess/filesource.rs

```diff
@@ -166,20 +167,11 @@ impl Write for TestWriterLock<'_> {
 }
 
 pub(crate) type TestWriterInner = Arc<Mutex<Vec<u8>>>;
-
 struct TestWriter(TestWriterInner);
 
-impl Writer for TestWriter {
-    fn lock(&self) -> Box<dyn WriterLock + '_> {
-        Box::new(TestWriterLock {
-            inner: self.0.lock().unwrap_or_else(|e| e.into_inner()),
-        })
-    }
-}
-
 impl Write for TestWriter {
     fn write(&mut self, buf: &[u8]) -> Result<usize> {
-        self.lock().write(buf)
+        self.0.lock().unwrap().write(buf)
```

Comment:
I'd -really- like us to avoid `unwrap()` where its not needed. Can't `?` be used?

Reply:
There is no from implementation for this error. So I use expect here.